cli/internal/cmd/init.go

```diff
@@ -252,7 +263,8 @@ func (d *initDoer) Do(ctx context.Context) error {
 	return nil
 }
 
-func (d *initDoer) logGRPCStateChanges(ctx context.Context, wg *sync.WaitGroup, conn *grpc.ClientConn) {
+func (d *initDoer) handleGRPCStateChanges(ctx context.Context, wg *sync.WaitGroup, conn *grpc.ClientConn) {
+	wg.Add(1)
 	go func() {
```

Comment:
I think this function would be a lot more readable if you moved the Go routine from being called in this function to the caller.
I.e  instead of doing `wg.Add(1)` and `go func() {` in `handleGRPCStateChanges`, do so in the calling function.
This would also have the benefit of making things a bit more clear to what is happening in the calling function.

Reply:
I personally agree with you but the others don‘t, so I don’t know ¯\\_(ツ)_/¯

However if we have this pattern already somewhere else in the code I would just go for consistency.

Reply:
@katexochen could you explain why you want to keep calling the go routine in the function?

Reply:
It bundles concurrency related code in a single place, which makes it easier to reason about. There is a single place where mistakes could be made. If the caller needs to handle it, you'd need to check every caller for correct use. Functions should always aim to hide complexity like this from the caller, same goes for synchronization primitives like mutex etc.